### PR TITLE
Fix/burn remote deadlock + burn fusion draining

### PR DESCRIPTION
**Pull request by nathanielsimard**
Improved a lot of things related to streams and training a model on multiple threads.

### CODE REVIEW

File: crates/burn-hip/src/lib.rs [hunk 1]
```diff
@@ -18,13 +18,15 @@ pub type Hip<F = f32, I = i32> = JitBackend<HipRuntime, F, I>;
...
+// #[cfg(target_os = "linux")]
+// #[cfg(test)]
+// mod tests {
+//     use burn_jit::JitBackend;
+//
+//     pub type TestRuntime = cubecl::hip::HipRuntime;
+//     pub use half::{bf16, f16};
+//
+//     burn_jit::testgen_all!();
+// }
```
Comment (laggui) [inline crates/burn-hip/src/lib.rs:32]: Not sure if this is meant to be commented out just locally or actually committed to fix some HIP stuff?

File: crates/burn-remote/src/client/worker.rs [hunk 2]
```diff
@@ -103,7 +103,9 @@ impl ClientWorker {
                     match msg {
                         Message::Binary(bytes) => {
                             let response: TaskResponse = rmp_serde::from_slice(&bytes).expect("Can deserialize messages from the websocket.");
+                            log::info!("On response lock");
                             let mut state = state_ws.lock().await;
+                            log::info!("On response");
```
Comment (laggui) [inline crates/burn-remote/src/client/worker.rs:108]: Debugging left overs or actual logs?

File: crates/burn-remote/src/client/worker.rs [hunk 3]
```diff
@@ -117,12 +119,14 @@ impl ClientWorker {
...
-                while let Some(req) = rec.recv().await {
+                while let Ok(req) = rec.recv().await {
                     let task = match req {
                         ClientRequest::WithSyncCallback(task, callback) => {
-                            let mut state = state.lock().await;
                             if let Task::Compute(_content, id) = &task {
+                            log::info!("With sync callback lock..");
+                                let mut state = state.lock().await;
                                 state.register_callback(*id, callback);
+                            log::info!("With sync callback done");
```
Comment (laggui) [inline crates/burn-remote/src/client/worker.rs:129]: Debugging left overs or actual logs?

File: examples/text-classification/Cargo.toml [hunk 4]
```diff
@@ -42,3 +42,4 @@ tokenizers = { version = "0.20.3", default-features = false, features = [
 # Utils
 derive-new = { workspace = true }
 serde = { workspace = true, features = ["std", "derive"] }
+log = { workspace = true }
```
Comment (laggui) [inline examples/text-classification/Cargo.toml:45]: Left over? I don't see it being used in the example.

**Review by laggui — APPROVED** [review]
Approving in advance, just have some minor comments that might need to be addressed before merging.

File: crates/burn-hip/src/lib.rs (hunk 1 above)
Comment (nathanielsimard) [reply crates/burn-hip/src/lib.rs:32]: It breaks rust-analyzer, so let's comment them until it's fixed.

**Pull request merged**